### PR TITLE
feature: multistage dockerfile for nginx with nap and agent

### PR DESCRIPTION
**Pull request by luk90**
part of ind-19134

NAP setup for ubuntu

### CODE REVIEW

**Review by azych — APPROVED** [review]

File: centos/nap/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    echo "stopping controller-agent ..."
+    kill -TERM "${agent_pid}" 2>/dev/null
+    echo "stopping nginx ..."
+    kill -TERM "${nginx_pid}" 2>/dev/null
+}
+
+trap 'handle_term' TERM
+
+# Launch NAP
+/bin/su -s /bin/bash -c '/opt/app_protect/bin/bd_agent &' nginx
```
Comment (apasyniuk) [inline centos/nap/entrypoint.sh:30]: You just run this in the background and do not handle any kind of graceful shutdown. Do that as in the case of agent or Nginx(same in all entrypoint versions)

File: debian/nap/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+test -n "${ENV_CONTROLLER_INSTANCE_NAME}" && \
+    instance_name=${ENV_CONTROLLER_INSTANCE_NAME}
+
+test -n "${ENV_CONTROLLER_URL}" && \
+    controller_url=${ENV_CONTROLLER_URL}
+
+test -n "${ENV_CONTROLLER_LOCATION}" && \
+    location=${ENV_CONTROLLER_LOCATION}
+
+if [ -n "${api_key}" -o -n "${instance_name}" -o -n "${controller_url}" -o -n "${location}" ]; then
```
Comment (apasyniuk) [inline debian/nap/entrypoint.sh:53]: Check for `controller_url` is not needed here I think.

File: debian/nap/nginx-plus-api.conf [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	#auth_basic_user_file /etc/nginx/users;
+
+	# Uncomment to use permissions based on IP address; see (4) above
+	#allow 10.0.0.0/8;
+	#deny all;
+
+	# Conventional location for accessing the NGINX Plus API
+	app_protect_enable on;
+	app_protect_security_log_enable on;
+    app_protect_security_log "/etc/nginx/custom_log_format.json" stderr;
```
Comment (apasyniuk) [inline debian/nap/nginx-plus-api.conf:75]: Indent.

**Review by apasyniuk — CHANGES_REQUESTED** [review]

**Review by apasyniuk — APPROVED** [review]

**Comment by 1996sajal** [comment]
Tests are failing on NAP docker containers when WAF_ENABLED.
`RuntimeError: The component entered an isError state while checking for configuration.See data provided in currentStatus.state.conditions: [{'message': 'The Security module could not be found on the following NGINX instances: i-0041bbb0960bae1b6', 'type': 'config'}]`

**Review by 1996sajal — CHANGES_REQUESTED** [review]
Tests are failing on NAP docker containers when WAF_ENABLED.
`RuntimeError: The component entered an isError state while checking for configuration.See data provided in currentStatus.state.conditions: [{'message': 'The Security module could not be found on the following NGINX instances: i-0041bbb0960bae1b6', 'type': 'config'}]`

**Review by 1996sajal — APPROVED** [review]
There have to be certain changes made to Systest common the way NAP tests are run and I think the MR can be merged to master as of now.

**Pull request merged**